### PR TITLE
fix: Try polling again on connection refused

### PR DESCRIPTION
**Pull request by Artur-**
This should fix problems with the dev mode not ready page showing forever on startup

### CODE REVIEW

**Review by czp13 — APPROVED** [review]

File: vaadin-dev-server/src/main/resources/com/vaadin/base/devserver/dev-mode-not-ready.html [hunk 1]
```diff
@@ -54,7 +54,7 @@
 const poll = () => {
   try {
     fetch(window.location.href, { headers: { 'X-DevModePoll': 'true' } }).then(response => {
```
Comment (platosha) [inline vaadin-dev-server/src/main/resources/com/vaadin/base/devserver/dev-mode-not-ready.html:56]: Needs an `await` to catch async errors in the outer `try ... catch`:
```suggestion
    await fetch(window.location.href, { headers: { 'X-DevModePoll': 'true' } }).then(response => {
```

File: vaadin-dev-server/src/main/resources/com/vaadin/base/devserver/dev-mode-not-ready.html [hunk 2]
```diff
@@ -54,7 +54,7 @@
...
-      if (response.headers.has("X-DevModePending")) {
+      if (!response || !response.headers || response.headers.has("X-DevModePending")) {
```
Comment (platosha) [inline vaadin-dev-server/src/main/resources/com/vaadin/base/devserver/dev-mode-not-ready.html:57]: Seems superficial, `fetch` must resolve to `Response` as per spec.

```suggestion
      if (response.headers.has("X-DevModePending")) {
```

Comment (Artur-) [reply vaadin-dev-server/src/main/resources/com/vaadin/base/devserver/dev-mode-not-ready.html:57]: Not sure what the case was there `then` was called with an undefined `response` but maybe it is not important

File: vaadin-dev-server/src/main/resources/com/vaadin/base/devserver/dev-mode-not-ready.html (hunk 1 above)
Comment (Artur-) [reply vaadin-dev-server/src/main/resources/com/vaadin/base/devserver/dev-mode-not-ready.html:56]: There is not top level await support in all Vaadin 23 browsers, so maybe `.catch()` is better then

Comment (Artur-) [reply vaadin-dev-server/src/main/resources/com/vaadin/base/devserver/dev-mode-not-ready.html:56]: ah but this isn't even top level...

Comment (Artur-) [reply vaadin-dev-server/src/main/resources/com/vaadin/base/devserver/dev-mode-not-ready.html]: Maybe the new version makes more senes

**Review by platosha — APPROVED** [review]

**Review by czp13 — APPROVED** [review]

**Pull request merged**